werkzeug/routing.py

```diff
@@ -1607,6 +1630,23 @@ def _partial_build(self, endpoint, values, method, append_unknown):
                 if rv is not None:
                     return rv
 
+    def _closest_rule(self, endpoint, values, method):
+        """Helper for :meth:`build`. Returns the closest rule the developer
+        is probably expecting given this endpoint, values and method.
+
+        :internal:
+        """
+        def score_rule(rule, endpoint, values, method):
```

Comment:
I get that adding ``endpoint, values, method`` params to this scoring func makes a cleaner API, but since it's neither exposed nor overridable, I think you should only accept ``rule`` here, to be able to avoid the additional lambda for ``max(...)``.